src/main/java/org/jenkinsci/plugins/orgfolder/github/GitHubLibraryResolver.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2016 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.orgfolder.github;
+
+import hudson.Extension;
+import hudson.model.Job;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import jenkins.plugins.git.GitSCMSource;
+import org.jenkinsci.plugins.workflow.libs.LibraryConfiguration;
+import org.jenkinsci.plugins.workflow.libs.LibraryResolver;
+
+/**
+ * Allows libraries to be loaded on the fly from GitHub.
+ */
+@Extension public class GitHubLibraryResolver extends LibraryResolver {
+
+    @Override public boolean isTrusted() {
+        return false;
+    }
+
+    @Override public Collection<LibraryConfiguration> forJob(Job<?,?> job, Map<String,String> libraryVersions) {
+        List<LibraryConfiguration> libs = new ArrayList<>();
+        for (Map.Entry<String,String> entry : libraryVersions.entrySet()) {
+            if (entry.getKey().matches("github[.]com/([^/]+)/([^/]+)")) {
```

Comment:
So this won't work for GitHub Enterprise?

Reply:
Would it be possible to re-use the credentials that are used for the job that's being run (e.g. in `checkout scm` and for fetching the Jenkinsfile) to fetch the shared library?

Let's say I have 2 private GitHub repositories, both requiring the same credentials for access. If Jenkins is already building one of the repositories (e.g. with GitHub Branch Source Plugin), it would be very convenient if `@Library('github.com/owner/other-repo) _` would just work for fetching shared code from the other repository.

I know this is old and the code has since been moved to https://github.com/jenkinsci/pipeline-github-lib-plugin, but this seemed an appropriate place for this question because credentials were mentioned in this thread.

Reply:
@deiwin comments to long-closed PRs are likely to get lost. Better to file an RFE in JIRA under `pipeline-github-lib-plugin`.

Reply:
Understood. Created https://issues.jenkins-ci.org/browse/JENKINS-49068